fault-quarantine/pkg/eventwatcher/event_watcher.go

```diff
@@ -23,27 +23,12 @@ import (
 
 	"github.com/nvidia/nvsentinel/data-models/pkg/model"
 	"github.com/nvidia/nvsentinel/fault-quarantine/pkg/metrics"
-	"github.com/nvidia/nvsentinel/store-client/pkg/storewatcher"
-	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/bson/primitive"
-	"go.mongodb.org/mongo-driver/mongo"
-	"go.mongodb.org/mongo-driver/mongo/options"
+	"github.com/nvidia/nvsentinel/store-client/pkg/client"
 )
 
-type MongoCollectionInterface interface {
-	UpdateOne(ctx context.Context, filter interface{}, update interface{},
-		opts ...*options.UpdateOptions) (*mongo.UpdateResult, error)
-	UpdateMany(ctx context.Context, filter interface{}, update interface{},
-		opts ...*options.UpdateOptions) (*mongo.UpdateResult, error)
-	FindOne(ctx context.Context, filter interface{}, opts ...*options.FindOneOptions) *mongo.SingleResult
-}
-
 type EventWatcher struct {
```

Comment:
not in scope for this PR but in the future it might be nice to have a generic EventWatcher or Reconciler that the watching controllers can share like fault-quarantine, fault-remediation, node-drainer, etc

Reply:
it'd be really cool if we could somehow get these event streams to work with controller-runtime reconciler pattern in the distant future